Source/WebCore/html/HTMLTextAreaElement.cpp

```diff
@@ -57,32 +57,15 @@ WTF_MAKE_ISO_ALLOCATED_IMPL(HTMLTextAreaElement);
 
 using namespace HTMLNames;
 
-// On submission, LF characters are converted into CRLF.
-// This function returns number of characters considering this.
-static unsigned computeLengthForSubmission(StringView text, unsigned numberOfLineBreaks)
-{
-    return text.length() + numberOfLineBreaks;
-}
-
-static unsigned numberOfLineBreaks(StringView text)
+static inline unsigned computeLengthForAPIValue(StringView text)
 {
     unsigned length = text.length();
-    unsigned count = 0;
-    for (unsigned i = 0; i < length; i++) {
-        if (text[i] == '\n')
-            count++;
+    unsigned crlfCount = 0;
+    for (unsigned i = 0; i < length; ++i) {
+        if (text[i] == '\r' && i + 1 < length && text[i + 1] == '\n')
```

Comment:
Not very important, but I think this would be slightly more readable if the loop condition was changed to `i < length - 1`, and the `if` statement was simplified.

Reply:
I like that idea! But if we literally did that, then we would break the zero-length string case. It needs to stay as `i + 1 < length` or some other idiom that does not underflow.

Reply:
But not only is that more readable, it’s likely a tiny bit more efficient as well. Since we are only searching for CRLF, we can do it this way.

Reply:
Good point – missed the `unsigned` 😅